File: fe_camphub/src/libs/types/Notification.ts
import { NotificationType, ReferenceType } from "../constants";

export interface Notification {
  id: string;                    // UUID
  receiverId: string;            // người nhận thông báo
  senderId?: string | null;      // người gửi (nếu có)
  type: NotificationType;        // loại thông báo
  title?: string | null;         // tiêu đề
  content?: string | null;       // nội dung
  referenceType?: ReferenceType | null; // loại thực thể liên quan (Booking, Item, v.v.)
  referenceId?: string | null;   // id liên quan
  isRead: boolean;               // đã đọc chưa
  createdAt: string;             // thời gian tạo
  readAt?: string | null;        // thời gian đọc (nếu có)
}

File: be_camphub/src/main/java/org/camphub/be_camphub/entity/Notification.java
package org.camphub.be_camphub.entity;

import java.time.LocalDateTime;
import java.util.UUID;

import jakarta.persistence.*;

import org.camphub.be_camphub.enums.NotificationType;
import org.camphub.be_camphub.enums.ReferenceType;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity
@Table(name = "notifications")
@FieldDefaults(level = AccessLevel.PRIVATE)
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Notification {
    @Id
    @GeneratedValue
    UUID id;

    @Column(name = "receiver_id", nullable = false)
    UUID receiverId;

    @Column(name = "sender_id")
    UUID senderId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    NotificationType type;

    @Column
    String title;

    @Column(columnDefinition = "TEXT")
    String content;

    @Enumerated(EnumType.STRING)
    @Column(name = "reference_type")
    ReferenceType referenceType;

    @Column(name = "reference_id")
    UUID referenceId;

    @Builder.Default
    @Column(name = "is_read")
    Boolean isRead = false;

    @Builder.Default
    @Column(name = "is_broadcast")
    Boolean isBroadcast = false;

    @CreatedDate
    @Column(name = "created_at")
    LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "read_at")
    LocalDateTime readAt;
}
